sdk/core/scheduler/main.cc

```diff
@@ -553,14 +599,45 @@ int futex_timed_wait(Timeout        *timeout,
 		return 0;
 	}
 	Thread *currentThread = Thread::current_get();
-	Debug::log("{} waiting on futex {} for {} ticks",
-	           currentThread,
+	Debug::log("Thread {} waiting on futex {} for {} ticks",
+	           currentThread->id_get(),
 	           address,
 	           timeout->remaining);
-	currentThread->futexWaitAddress = Capability{address}.address();
+	bool      isPriorityInheriting         = flags & FutexPriorityInheritance;
+	ptraddr_t key                          = Capability{address}.address();
+	currentThread->futexWaitAddress        = key;
+	currentThread->futexPriorityInheriting = isPriorityInheriting;
+	Thread  *blockingThread                = nullptr;
```

Comment:
🚲🏚️ing: Maybe `owning` rather than `blocking`?  Yes, that thread is blocking our progress, so I understand, but also *we're* about to block on that thread's ownership of this futex.  (Naming things is _hard_!)